app/controllers/content_items_controller.rb

```diff
@@ -7,6 +7,9 @@ class ContentItemsController < ApplicationController
 
   def index
     @content_items = ContentItemsQuery.build_paged(@query).decorate
+
+    unpaged_content_items = ContentItemsQuery.build(@query)
+    @metrics = MetricBuilder.new.run_collection(unpaged_content_items)
```

Comment:
I would love to be able to express this method this way:

```ruby
def index
  @content_items = ContentItemsQuery.build_paged(@query).decorate
  @collection_metrics = MetricBuilder.new.run_collection(@query)
end
```

Ideally the controller should not know what the `MetricBuilder` object needs. If we pass the query, when running all collection metrics, then at that point, we now that we should not paginate the results.

It might be that we need to move into something like:
```ruby
def index
  query = ContentItemsQuery.new(options)
  @content_items = query.build_paginated
  @metrics = MetricBuilder.new.run_collection(query)
end
```

Let's have a quick chat about it!